test/queryinterface.spec.js

```diff
@@ -331,15 +385,18 @@ describe('QueryInterface', function () {
 				qi = new QueryInterface();
 			});
 			
-			it('should default to calling into parent queue if a parent exists', function () {
+			it('should default to calling into parent queue if a parent exists', function (done) {
 				var run = 0;
 				qi._results = [];
 				qi.options.parent = {
 					$query: function () { run++; return 'foo'; }
 				};
 				
-				qi.$query().should.equal('foo');
-				run.should.equal(1);
+				qi.$query().then(function (content) {
```

Comment:
I had to change this tests because it was "broken". It was testing the functionality but breaking the contract of `$query()` because the real `$query()` returns a Promise. Same for the tests below.